### PR TITLE
Use eshell for dpkg with readline interface

### PR DESCRIPTION
**Pull request by dvzubarev**
Dpkg provides many interfaces for interacting with an user. It is customized via
DEBIAN_FRONTEND environment variable. The interface that is used by default
doesn't work well with eshell. So the interface is switched to readline and
added an option to disable user prompts
entirely (helm-system-packages-dpkg-confirm-p).

I tested this PR on ubuntu 16.04 and debian 8.5.

### CODE REVIEW

File: helm-system-packages-dpkg.el [hunk 1]
```diff
@@ -308,18 +281,26 @@ If REVERSE is non-nil, show reverse dependencies instead."
                      (mapconcat 'identity (helm-marked-candidates) " "))))
         (helm-system-packages-dpkg)))))
 
+(defun helm-system-packages-make-apt-get-command ( &rest args)
+  (let ((comm (append '("apt-get" "-q") args)))
+    (if helm-system-packages-dpkg-confirm-p
+        (append '("DEBIAN_FRONTEND=readline") comm)
+      (append '("DEBIAN_FRONTEND=noninteractive") comm))))
```
Comment (Ambrevar) [inline helm-system-packages-dpkg.el]: `push` would be easier here I think.

File: helm-system-packages-dpkg.el [hunk 2]
```diff
@@ -308,18 +281,26 @@ If REVERSE is non-nil, show reverse dependencies instead."
                      (mapconcat 'identity (helm-marked-candidates) " "))))
         (helm-system-packages-dpkg)))))
 
+(defun helm-system-packages-make-apt-get-command ( &rest args)
+  (let ((comm (append '("apt-get" "-q") args)))
```
Comment (Ambrevar) [inline helm-system-packages-dpkg.el]: Please use long options (here `--quiet`) as they make the code more explicit and readable.
Short options are useful for interactive use.

File: helm-system-packages-dpkg.el [hunk 3]
```diff
@@ -308,18 +281,26 @@ If REVERSE is non-nil, show reverse dependencies instead."
...
   '(("Show package(s)" .
      (lambda (_)
        (helm-system-packages-print "apt-cache" "show")))
     ("Install (`C-u' to reinstall)" .
      (lambda (_)
-       (helm-system-packages-dpkg-run-as-root
-        "apt-get" "install" (when helm-current-prefix-arg "--reinstall"))))
+       (apply 'helm-system-packages-run-as-root
+              (helm-system-packages-make-apt-get-command "install"
+                                                         (when helm-current-prefix-arg "--reinstall")))))
```
Comment (Ambrevar) [inline helm-system-packages-dpkg.el:298]: Why not keeping the command as it was and insert

```
  (if helm-system-packages-dpkg-confirm-p
	    "DEBIAN_FRONTEND=readline"
    "DEBIAN_FRONTEND=noninteractive")
```

before "apt-get"?

File: helm-system-packages-dpkg.el [hunk 4]
```diff
@@ -332,8 +313,9 @@ If REVERSE is non-nil, show reverse dependencies instead."
        (helm-system-packages-print "apt-cache" "rdepends")))
     ("Uninstall/Purge (`C-u' to include dependencies)" .
      (lambda (_)
-       (helm-system-packages-dpkg-run-as-root
-        "apt-get" "purge" (when helm-current-prefix-arg "--auto-remove")))))
+       (apply 'helm-system-packages-run-as-root
+              (helm-system-packages-make-apt-get-command "purge"
+                                                         (when helm-current-prefix-arg "--auto-remove"))))))
```
Comment (Ambrevar) [inline helm-system-packages-dpkg.el:318]: Same here.

File: helm-system-packages-dpkg.el (hunk 1 above)
Comment (Ambrevar) [reply helm-system-packages-dpkg.el]: This function could go altogether I think.  See below.

File: helm-system-packages-dpkg.el (hunk 3 above)
Comment (dvzubarev) [reply helm-system-packages-dpkg.el:298]: I don't like idea to duplicate this piece of code three times and maybe more if some new actions will be added later. May be we can get rid of apply somehow to make this simpler.

Comment (Ambrevar) [reply helm-system-packages-dpkg.el:298]: Fair enough, then go ahead and I'll merge your changes.  Thanks a lot!

**Comment by Ambrevar** [comment]
Another nit: please prefix the commit message with the package manager, i.e.

> dpkg: Use Eshell with readline interface

**Comment by Ambrevar** [comment]
Excellent!  Thank you very much!

**Pull request merged**